src/unitxt/operators.py

```diff
@@ -376,6 +376,67 @@ def process_value(self, value: Any) -> Any:
         return new_value
 
 
+class AugmentSuffix(Augmentor):
+    """
+    Augments the input by appending to it a randonly selected (typically, whitespace) pattern.
+    Any existing trailig whitespace (if any), of the input is first discarded.
+    Then, a pattern is randomly selected from a given list of (pattern, weight) pairs.
+    Finally, the selected pattern is appended to the (potentially trimmed at its end) input.
+
+    Args:
+     suffixes : the potential white-space patterns to select from.
+        the dictionary version allowes to specify relative weights of the different patterns.
+
+    Examples:
+        to append a '\n' or a '\t' to the end of the input, employ
+        AugmentSuffix(augment_model_input=True, suffixes=['\n','\t'])
+
+    """
+
+    suffixes: Optional[Union[List[str], Dict[str, int]]] = [" ", "\n", "\t"]
+
+    def verify(self):
+        assert isinstance(self.suffixes, list) or isinstance(
+            self.suffixes, dict
+        ), f"Argument 'suffixes' should be either a list or a dictionary, whereas it is of type {type(self.suffixes)}"
+
+        if isinstance(self.suffixes, dict):
+            for k, v in self.suffixes.items():
+                assert isinstance(k, str), f"suffixes should map strings, whereas key {str(k)} is of type {type(k)}"
+                assert isinstance(v, int), f"suffixes should map to ints, whereas value {str(v)} is of type {type(v)}"
+        else:
+            for k in self.suffixes:
+                assert isinstance(
+                    k, str
+                ), f"suffixes should be a list of strings, whereas member {str(k)} is of type {type(k)}"
+
+        super().verify()
+        self.next_pats = self.select_next_100_patterns()
+
+    def process_value(self, value: Any) -> Any:
+        assert value is not None, "input value should not be None"
+        new_value = str(value).rstrip()
+        assert len(new_value) > 0, f"nothing left of input value {str(value)} after trimming trailing whitespaces"
+        if len(self.next_pats) == 0:
+            self.next_pats = self.select_next_100_patterns()
```

Comment:
Why not select one element at a time?

Reply:
I am not sure when the seed for the random is set. If it is set to same value each time this method is entered, then the same random result is output.

Reply:
The random set is set for every value that is being processed:


     with nested_seed(str(hash(old_value))):
                try:
                    new_value = self.process_value(old_value)
                except Exception as e:
                
 So it's ok.    
 
 You can calculate the weights only once.

Seems like simpler code.

Reply:
Hi.  You don't need to add the nested_seed wrapper, as it is handled in the parent class.